### PR TITLE
RDK-29310 : Remote Control Passthrough

### PR DESCRIPTION
**Pull request by bijas**
Reason for change: Implementation of remote control pass through for VOLUME_UP,VOLUME_DOWN,MUTE of arc device for hdmicecsink
Test Procedure: none
Risks: low

Signed-off-by: Bijas Babu bijas.babu@sky.uk

### CODE REVIEW

File: HdmiCecSink/HdmiCecSink.cpp [hunk 1]
```diff
@@ -110,7 +113,8 @@ static char *eventString[] = {
         "arcTerminationEvent",
         "shortAudiodesciptorEvent",
         "standbyMessageReceived",
-        "setSystemAudioModeEvent"
+        "setSystemAudioModeEvent",
```
Comment (apatel859) [inline HdmiCecSink/HdmiCecSink.cpp:116]: reportAudioStatusEvent is defined in user sotry

File: HdmiCecSink/HdmiCecSink.cpp [hunk 2]
```diff
@@ -888,7 +898,39 @@ namespace WPEFramework
...
+			LOGINFO("Command: ReportAudioStatus  %s audio Mute status %d  means %s  and current Volume level is %d \n",GetOpName(msg.opCode()),msg.status.getAudioMuteStatus(),msg.status.toString().c_str(),msg.status.getAudioVolume());
+            params["muteStatus"]  = msg.status.getAudioMuteStatus();
+            params["volumeLevel"] = msg.status.getAudioVolume();
+            sendNotify(eventString[HDMICECSINK_EVENT_REPORT_AUDIO_STATUS], params);
 
+         }
+		 void HdmiCecSink::sendKeyPressEvent(const int logicalAddress, int keyCode)
+		 {
+		    LOGINFO(" sendKeyPressEvent logicalAddress 0x%x keycode 0x%x\n",logicalAddress,keyCode);
+                    switch(keyCode)
```
Comment (apatel859) [inline HdmiCecSink/HdmiCecSink.cpp:915]: No need to define VOLUME_UP VOLUME_DOWN etc as we are anyway asking UICommand::UI_COMMAND_VOLUME_UP from UI so below three switch cased can be clubbed in one

File: HdmiCecSink/HdmiCecSink.cpp (hunk 1 above)
Comment (bijas) [reply HdmiCecSink/HdmiCecSink.cpp:116]: "setSystemAudioModeEvent",
        "reportAudioStatus"
};

this is from the above file

File: HdmiCecSink/HdmiCecSink.cpp [hunk 3]
```diff
@@ -903,7 +945,14 @@ namespace WPEFramework
            _instance->smConnection->sendTo(LogicalAddress::AUDIO_SYSTEM,MessageEncoder().encode(SystemAudioModeRequest(physical_addr)), 1100);
 
         }
+         void HdmiCecSink::sendgiveAudioStatusMsg()
```
Comment (apatel859) [inline HdmiCecSink/HdmiCecSink.cpp:948]: typo sendG

**Review by apatel859 — APPROVED** [review]

**Pull request merged**